selftests/functional/test_basic.py

```diff
@@ -106,7 +108,13 @@ def probe_binary(binary):
 
 
 CC_BINARY = probe_binary('cc')
-ECHO_BINARY = probe_binary('echo')
+# On macOS, the default coreutils installation (brew)
+# installs the gnu utility versions with a g prefix.
+# First, look if there is a gecho, use that if that's the case
+# This won't hurt linux systems (no need for prefixing coreutils with g)
+ECHO_BINARY = probe_binary('gecho')
+if ECHO_BINARY is None:
+    ECHO_BINARY = probe_binary('echo')
```

Comment:
The problem is that macOS has its own, BSD version of `echo`, which does not have the `-e` flag, a fundamental part of what makes the avocado functional test to work. So probing for echo first leads to this version being selected, and the test will fail.

So probing first for `gecho` is most likely harmless, while making things just work on both linux and macOS systems that have the gnu coreutils installed with the `g` prefix.

The thing with `echo` is that the only way to know whether it supports the `-e` option is looking at its man page and performing a regexp search for said option. I wanted to keep it simple, so I went with this "slightly less intuitive" construct.

If you tell me that I should go ahead and start looking into the man pages for the options, then... sure.